src/PoolManager.sol

```diff
@@ -311,12 +311,18 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
     }
 
     function extsload(bytes32 startSlot, uint256 nSlots) external view returns (bytes memory) {
-        bytes memory value = new bytes(32 * nSlots);
+        // For values of nSlots that cause an overflow, an out-of-gas exception will be thrown.
+        bytes memory value = new bytes(nSlots << 5);
```

Comment:
```Solidity
        /// @solidity memory-safe-assembly
        assembly {
            let endSlot := add(startSlot, nSlots)
            let valuePtr
            for {} 1 {} {
                mstore(valuePtr, sload(startSlot))
                valuePtr := add(valuePtr, 0x20)
                startSlot := add(startSlot, 1)
                if iszero(lt(startSlot, endSlot)) { break }
            }
            // For values of nSlots that cause an overflow, an out-of-gas exception will be thrown.
            return(0, shl(5, nSlots))
        }
```
Because `extsload` is an external function that enters a new call context and exits right after execution, we can disregard Solidity's memory management convention and directly return a slice of memory.